Doc/library/decimal.rst

```diff
@@ -598,6 +598,23 @@ Decimal objects
 
       .. versionadded:: 3.1
 
+   .. classmethod:: from_number(number)
+
+      Alternative constructor that only accepts numbers (instances of
```

Comment:
Instead of using `number`, maybe you can directly say `only accepts instances of float, int or Decimal) since otherwise people might wonder why objects implementing the Number protocol are not allowed.

By the way, this function is essentially a shortcut to avoid an `if isinstance(...)` right?

Reply:
Done.

Yes, you can look at it from such point. If your function needs a Decimal or a _number_ that can be converted to Decimal, it can use `Decimal.from_number()` without additional type check. It will reject strings and tuples that are accepted by the constructor.